src/txdb.cpp

```diff
@@ -35,21 +33,25 @@ CCoinsViewDB::CCoinsViewDB(size_t nCacheSize, bool fMemory, bool fWipe) : db(Get
 }
 
 bool CCoinsViewDB::GetCoins(const uint256 &txid, CCoins &coins) const {
+    boost::shared_lock<boost::shared_mutex> lock(utxo);
```

Comment:
It's a good question...I think i was debugging a problem and put an extra layer of isolation.  I don't think I trust LevelDB entirely so I think I'd like to leave it in place for a little while.  It can probably be taken out but as I said it's a low cost extra layer of isolation which just gives me confidence that when/if I see a problem  then I know it's not  levelDB.